src/test/spec/unified_runner/test_file.rs

```diff
@@ -145,15 +145,38 @@ pub struct Client {
     pub use_multiple_mongoses: Option<bool>,
     pub observe_events: Option<Vec<String>>,
     pub ignore_command_monitoring_events: Option<Vec<String>>,
-    pub observe_sensitive_commands: Option<bool>,
     #[serde(default)]
+    pub observe_sensitive_commands: Option<bool>,
+    #[serde(default, deserialize_with = "deserialize_server_api")]
     pub server_api: Option<ServerApi>,
 }
 
 fn default_uri() -> String {
     DEFAULT_URI.clone()
 }
 
+pub fn deserialize_server_api<'de, D>(
```

Comment:
Why was this custom implementation needed?

Reply:
The server expects the fields to be named `apiVersion`, `apiStrict`, and `apiDeprecationErrors`, but the test format drops all of the `api-` prefixes. Now that we're using the `Serialize` implementation for `ServerApi` in the driver, we needed to introduce a custom one for deserializing from the test format.

Reply:
I know this is only used within this test file, but for clarity can we rename this serialization function? Maybe `deserialize_server_api_for_testing` or `deserialize_server_api_test_format`?

Reply:
Ah in that case I think you should be able to change the `rename` attributes on the fields in `ServerApi` to `#[serde(rename(serialize = "apiVersion"))]` so that the renames only apply to serialization, and then add `#[serde(rename_all = "camelCase", deny_unknown_fields)]` to the top of the struct for deserialization.

Reply:
Oh cool, I didn't know that was possible. Unfortunately, it looks like we'll still need the separate implementation though, since `deny_unknown_fields` can't be used with `flatten` (see https://serde.rs/attr-flatten.html). `flatten` is needed with server API since they're all top level fields of commands rather than in a subdocument.

I did update the fuinction to be more precisely named (went with the `test_format` suggestion)